docs/language_concepts/00_data_types.md

```diff
@@ -403,3 +403,27 @@ fn multiplyBy2(x: &mut Field) {
     *x = *x * 2;
 }
 ```
+
+## Function Types
+
+Noir supports higher-order functions. The syntax for a function type is as follows:
+
+```rust
+fn(arg1_type, arg2_type, ...) -> return_type
+```
+
+Example:
+
+```rust
+fn assert_returns_100(f: fn() -> Field) { // f takes no args and returns a Field
+    assert(f() == 100);
+}
+
+fn main() {
+    assert_returns_100(|| 100); // ok
+    assert_returns_100(|| 150); // fails
+}
+```
+
+A function type also has an optional capture environment - this is necessary to support closures.
+See [Lambdas](./lambdas) for more details.
```

Comment:
Shouldn't this be `(./08_lambdas.md)`?

Reply:
fixed: https://github.com/noir-lang/docs/pull/356/commits/df0de5f7abc7bfc7ffca7a1944c62e433cc81ba9